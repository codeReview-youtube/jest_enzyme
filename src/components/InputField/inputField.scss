.input {
  border: 0;
  outline: none;
  width: 100%;
  font-size: 2em;
  padding: 0.3em;
  border-radius: 0.2em;
  margin-bottom: 0.2em;
}
