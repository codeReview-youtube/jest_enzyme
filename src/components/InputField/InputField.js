import React from 'react';
import './inputField.scss';
/**
 * @author
 * @function InputField
 **/

const InputField = ({ onChange, value, onSubmit }) => {
  return (
    <form
      onSubmit={e => {
        e.preventDefault();
        onSubmit();
      }}>
      <input
        onChange={e => onChange(e.target.value)}
        value={value}
        type='text'
        className='input'
        placeholder='Add new ...'
      />
    </form>
  );
};

export default InputField;
