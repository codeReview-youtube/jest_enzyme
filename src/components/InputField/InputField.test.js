/* eslint-disable react/display-name */
import React from 'react';
import { shallow } from 'enzyme';
/**
 *import  Component
 */
import InputField from './InputField';
/**
 * props
 */
const props = {
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  value: 'todo'
};
/**
 * Tests: InputField Component
 */
describe('InputField Component', () => {
  /**
   * Snapshot:[default]
   */
  test('should match default snapshot', () => {
    expect.assertions(1);

    const wrapper = shallow(<InputField {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  /**
   * triggers [onSubmit]
   */
  test('should submit form', () => {
    expect.assertions(1);

    const wrapper = shallow(<InputField {...props} />);

    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(props.onSubmit).toHaveBeenCalled();
  });
  /**
   * triggers [onChange]
   */
  test('should submit form', () => {
    expect.assertions(1);
    const value = 'any';
    const wrapper = shallow(<InputField {...props} />);

    wrapper.find('input').simulate('change', {
      target: { value }
    });
    expect(props.onChange).toHaveBeenCalledWith(value);
  });
});
