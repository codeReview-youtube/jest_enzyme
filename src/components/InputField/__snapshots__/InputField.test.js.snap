// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component should match default snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Add new ..."
    type="text"
    value="todo"
  />
</form>
`;
